### ISSUE TITLE
How can I set X-opaque-Id to header  when searching?

### ISSUE THREAD

**Issue by ryook** [elastic/elasticsearch-py#847]
I want to set X-opaque-Id to header to track task.
There is API in elasticsearch, Can't I use it using this library?

https://www.elastic.co/guide/en/elasticsearch/reference/current/tasks.html#_identifying_running_tasks

**Comment by abhishek063** [comment]
+1. I was wondering the same. How to set this in python elasticsearch library?

**Comment by e9** [comment]
```
es = Elasticsearch(["http://localhost:9200"], headers={"x-opaque-id": "123456"})
res = es.search(...)
```

**Comment by sethmlarson** [comment]
This has been fixed in #1135, can be set per API request or on the client like so:
This will be available in `elasticsearch>=7.6.0`.

```python
es = Elasticsearch(..., opaque_id="app-1")
es.search(..., opaque_id="request-2")
```
